src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -45,6 +49,23 @@ private static bool GetIsStreamingSupported()
                 return (bool)streamingSupported.Call();
         }
 
+        private void CheckDisposed()
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException(nameof(BrowserHttpHandler));
+            }
+        }
+
+        private void CheckDisposedOrStarted()
+        {
+            CheckDisposed();
+            if (_handlerStarted)
```

Comment:
Why is this check needed and what are we protecting from?

Reply:
Similar check is in [`SocketsHttpHandler`](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L32).
The whole point is to prevent any changes to the handler settings after the first request has been sent.
Whether this is necessary for Browser handler to work properly is a different question though.

Reply:
Is this SocketsHttpHandler limitation or is the expectation that any handler cannot by tweaked after its first use ?

Reply:
I think it is a design point of `SocketsHttpHandler` - @geoffkizer @scalablecory will know for sure.
It would feel a bit weird if our handlers choose entirely different design points IMO ...

Reply:
It prevents races in Sockets handler. It's probably best to copy that pattern here for consistency, unless we have a good reason to do differently here.

Reply:
I think it's because we're setting up handler chain with the first request: 
https://github.com/kjpou1/runtime/blob/wasm-http-redirect/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/SocketsHttpHandler.cs#L435
After that, we reuse the created `_handler` so the tweaks to the settings wouldn't get propagated. It's probably better to let the user know that later settings changes are in vain than let him painfully debug why it's behaving differently than what they're expecting.

Reply:
The code discussed above was removed in the recent commits.  Thanks for the feedback everyone.